src/tool/tool.go

```diff
@@ -70,17 +57,26 @@ func matchingTools(config *core.Configuration, prefix string) map[string]string
 		"pex":         config.Python.PexTool,
 		"sandbox":     "please_sandbox",
 	}
+}
+
+// matchingTools returns a set of matching tools for a string prefix.
+func matchingTools(config *core.Configuration, prefix string) map[string]string {
 	ret := map[string]string{}
-	for k, v := range knownTools {
+	for k, v := range knownTools(config) {
 		if strings.HasPrefix(k, prefix) {
 			ret[k] = v
 		}
 	}
 	return ret
 }
 
-// allToolNames returns the names of all available tools.
-func allToolNames(config *core.Configuration, prefix string) []string {
+func MatchingTool(config *core.Configuration, tool string) (string, bool) {
+	tool, ok := knownTools(config)[tool]
+	return tool, ok
+}
+
+// AllToolNames returns the names of all available tools.
+func AllToolNames(config *core.Configuration, prefix string) []string {
```

Comment:
Does this need to be exported?

Reply:
You're right I don't think it does. It was being used for the unknown tool error message before so I've moved the error message back into tool.go